Upstream-Status: Backport

(reworked for M54)

Signed-off-by: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
---
From c4e3e52dff0f5b9dff7167a1564b0e1978d3ea2e Mon Sep 17 00:00:00 2001
From: "raphael.kubo.da.costa" <raphael.kubo.da.costa@intel.com>
Date: Tue, 16 Aug 2016 08:26:48 -0700
Subject: [PATCH] gn: Use a separate flag for enabling libgnome-keyring
 support.

Right now, |is_desktop_linux| means GN requires libgnome-keyring to
be present during the build and also at run-time.

Since libgnome-keyring has been deprecated by libsecret, is used by a
small number of users and is not always present on a Linux
distribution (especially embedded ones), it makes sense to introduce a
specific argument, |use_gnome_keyring|, to control whether the code
should be used or not.

At the moment, |use_gnome_keyring| defaults to on under pretty much all
circumstances where checking for |is_desktop_linux| would work before:
it just needs |use_glib| to be on as well, as required by
libgnome-keyring itself.

R=brettw@chromium.org,vabr@chromium.org,thestig@chromium.org,jochen@chromium.org
BUG=466975,602624

Review-Url: https://codereview.chromium.org/2244653002
Cr-Commit-Position: refs/heads/master@{#412246}
---
 chrome/browser/BUILD.gn          |  3 ++-
 chrome/test/BUILD.gn             |  3 ++-
 components/os_crypt/BUILD.gn     | 17 ++++++++++-------
 components/os_crypt/features.gni | 11 +++++++++++
 tools/gn/docs/cookbook.md        |  2 +-
 5 files changed, 26 insertions(+), 10 deletions(-)
 create mode 100644 components/os_crypt/features.gni

diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
index e504c2b..71641f1 100644
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -57,7 +57,7 @@ if (is_win) {
   }
 }
 
-if (is_desktop_linux) {
+if (use_gnome_keyring) {
   # Gnome-keyring is normally dynamically loaded. The gnome_keyring config
   # will set this up.
   pkg_config("gnome_keyring") {
@@ -573,7 +573,7 @@ source_set("browser") {
   if (use_cups) {
     configs += [ "//printing:cups" ]
   }
-  if (is_desktop_linux) {
+  if (use_gnome_keyring) {
     sources += rebase_path(gypi_values.chrome_browser_gnome_keyring_sources,
                            ".",
                            "//chrome")
diff --git a/chrome/common/features.gni b/chrome/common/features.gni
index 5cc2a09..09c01f0 100644
--- a/chrome/common/features.gni
+++ b/chrome/common/features.gni
@@ -5,6 +5,8 @@
 # This is the GN version of //chrome/chrome_features.gypi.
 # Please keep in sync!
 
+import("//build/config/ui.gni")
+
 declare_args() {
   enable_google_now = !is_ios && !is_android
 
@@ -26,6 +28,8 @@ declare_args() {
   # Set to true to bundle all the mash related mojo services into chrome.
   # Specify --mash to chrome to have chrome start the mash environment.
   enable_package_mash_services = is_chromeos
+
+  use_gnome_keyring = is_desktop_linux && use_glib
 }
 
 chrome_grit_defines = [
diff --git a/chrome/test/BUILD.gn b/chrome/test/BUILD.gn
index 38a0731..bfdd899 100644
--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -2032,7 +2032,7 @@ test("unit_tests") {
   } else {
     sources -= [ "../browser/password_manager/password_store_x_unittest.cc" ]
   }
-  if (is_desktop_linux && current_cpu == "x64") {
+  if (use_gnome_keyring && current_cpu == "x64") {
     # Only add this test for 64 bit builds because otherwise we need the 32
     # bit library on 64 bit systems when running this test.
     sources +=
-- 
2.7.4

